clippy_utils/src/lib.rs

```diff
@@ -326,6 +326,25 @@ pub fn is_trait_method(cx: &LateContext<'_>, expr: &Expr<'_>, diag_item: Symbol)
         .map_or(false, |did| is_diag_trait_item(cx, did, diag_item))
 }
 
+/// Checks if the given expression is a path referring an item on the trait
+/// that is marked with the given diagnostic item.
+///
+/// For checking method call expressions instead of path expressions, use
+/// [`is_trait_method`].
+///
+/// For example, to find if an expression like `u64::default` refers to an item
+/// of the trait `Default`, which is marked `#[rustc_diagnostic_item = "Default"]`,
+/// a `diag_item` of `sym::Default` should be used.
```

Comment:
I like the fact that you're also illustrating with an example what this function is actually checking. But the second part could be a bit confusing. We should document how diagnostic items can be found and added, but this is not the right place IMO. I'll open an issue for that in the [rustc-dev-guide](https://github.com/rust-lang/rustc-dev-guide) repo.

```suggestion
/// For example, it can be used to check if an expression like `Type::default` refers to
/// an item of the `Default` trait or an item of the type.
```

This update can be done in an additional commit and doesn't have to be squashed. Then it's ready for the merge :upside_down_face: